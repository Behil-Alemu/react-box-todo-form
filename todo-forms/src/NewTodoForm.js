import React, { useState } from 'react';
import { v4 as uuid } from 'uuid';

const NewListItemForm = ({ addTodo }) => {
	const [ task, setTask ] = useState('');

	const handleChange = (evt) => {
		const { value } = evt.target;
		setTask(value);
	};

	const handleSubmit = (evt) => {
		evt.preventDefault();
		addTodo({ task, id: uuid() });
		setTask('');
	};

	return (
		<form onSubmit={handleSubmit}>
			<label htmlFor="task">My Tasks:</label>
			<input id="task" placeholder="TO-DO" type="text" name="task" value={task} onChange={handleChange} />
			<button className="addTaskBtn">Add a new tasks!</button>
		</form>
	);
};

export default NewListItemForm;
