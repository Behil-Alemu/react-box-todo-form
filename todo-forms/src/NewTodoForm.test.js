import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import NewTodoForm from "./NewTodoForm";

it("renders without crashing", function() {
  render(<NewTodoForm />);
});

it("matches snapshot", function() {
  const { asFragment } = render(<NewTodoForm />);
  expect(asFragment()).toMatchSnapshot();
});

// it("runs the create function on form submit", function() {
//   const button = screen.getByText("Add a new tasks!");
//   fireEvent.click(button);

// });