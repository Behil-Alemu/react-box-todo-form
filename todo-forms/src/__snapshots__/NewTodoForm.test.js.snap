// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="task"
    >
      My Tasks:
    </label>
    <input
      id="task"
      name="task"
      placeholder="TO-DO"
      type="text"
      value=""
    />
    <button
      class="addTaskBtn"
    >
      Add a new tasks!
    </button>
  </form>
</DocumentFragment>
`;
